requirements.txt

```diff
@@ -0,0 +1,11 @@
+jsonschema~=4.7
+lxml>=4.2,<5
+python-dateutil>=2.8,<3.0
+types-python-dateutil
+pyecma376-2>=0.2.4
+urllib3>=1.26,<2.0
+Werkzeug>=3.0.3,<4
+schemathesis~=3.7
+hypothesis~=6.13
+lxml-stubs~=0.5.1
+aas-core3.0~=1.0.4
```

Comment:
This file is not directly necessary for the sphinx compilation but rather a way to install all required packages in the CI-jobs context. You are right, we should definetely achieve this by using the pyproject.toml and adding everything .
However there is also currently the 'add-requirements.txt' which is only used by the docs job to install the additional dependencies necessary for compilation. This one I would keep as its a pretty neat way to add all the imports.
I also just noticed that i still have to check wether every of these entries is acutally necessary.

Reply:
Okay I just rechecked and it is indeed necessary to have at least one requirements.txt file that is passed to readthedocs for the compilation of the sphinx docs on their side.
We could definetely achieve a structure only based on the pyproject.toml by defining an additional, optional, dependency set like so:
```
[project.optional-dependencies]
docs = [
  "sphinx",
  "sphinx-rtd-theme",
  "sphinx-argparse",
  "sphinx_autodoc_typehints",
  "basyx-python-sdk"
] 
```
and adjusting the imports used in the github CI. But I don't see a way this could be passed to RTD [[https://docs.readthedocs.io/en/stable/guides/reproducible-builds.html]].

I think our best option would be to drop the toplevel 'requirements.txt' and only keep one; containing the dependency AND additional dependencies.

But then we would have to declare dependencies in the pyproject.toml as well as in the requirements file which I am not really a fan of. Maybe we could generate the requirements file to always display whatever is declared in the pyproject.toml dependencies AND optional dependencies? Or are we just fine with declaring every dependency at two different points?

Reply:
Let's talk about this in person when we next meet.

Reply:
Okay so as discussed in our meeting I restructured the dependency specifications.
Unfortunately I was not able to verify RTD compilation so I cant tell wether the current setup would function and wether every necessary dependency is listed. Maybe this will have to be changed later on (again).